packages/editor/src/components/rich-text/index.js

```diff
@@ -738,13 +743,66 @@ export class RichText extends Component {
 		}
 	}
 
+	/**
+	 * Applies given annotations to the given record.
+	 *
+	 * @param {Object} record The record to apply annotations to.
+	 * @param {Array} annotations The annotation to apply.
+	 * @return {Object} A record with the annotations applied.
+	 */
+	applyAnnotations( record, annotations ) {
```

Comment:
This whole method could be extracted out of this file and covered with tests. It's a relatively complex logic.

Reply:
Could this file be moved to the annotation format in the format library? It is annotation specific.